lvsfunc/comparison.py

```diff
@@ -143,60 +96,53 @@ class Stack(Comparer):
     """
 
-    def __init__(self, direction: Direction,
-                 label_alignment: Optional[Literal[1, 2, 3, 4, 5, 6, 7, 8, 9]] = None,
-                 **clips: vs.VideoNode):
+    def __init__(self, clips: Union[Dict[str, vs.VideoNode], Sequence[vs.VideoNode]],
+                 /, *,
+                 direction: Direction = Direction.HORIZONTAL,
+                 label_alignment: int = 7
+                 ) -> None:
         self.direction = direction
-
-        if label_alignment is None and self.direction == Direction.VERTICAL:
-            label_alignment = 8
-        elif label_alignment is None and self.direction == Direction.HORIZONTAL:
-            label_alignment = 7
-
-        super().__init__(label_alignment, **clips)
+        super().__init__(clips, label_alignment=label_alignment)
 
     def _compare(self) -> vs.VideoNode:
+        self._markclips()
+
         if self.direction == Direction.HORIZONTAL:
             if not self.height:
-                raise ValueError('StackHorizontal requires that all clips be the same height.')
-            return core.std.StackHorizontal(self._markclips())
+                raise ValueError("Stack: StackHorizontal requires that all clips be the same height")
+            return core.std.StackHorizontal(self.clips)
 
-        elif self.direction == Direction.VERTICAL:
-            if not self.width:
-                raise ValueError('StackVertical requires that all clips be the same width.')
-            return core.std.StackVertical(self._markclips())
+        if not self.width:
+            raise ValueError("Stack: StackVertical requires that all clips be the same width")
+        return core.std.StackVertical(self.clips)
 
 
 class Interleave(Comparer):
     """
     From the VapourSynth documentation: Returns a clip with the frames from all clips interleaved.
     For example, Interleave(A=clip1, B=clip2) will return A.Frame 0, B.Frame 0, A.Frame 1, B.Frame 1, ...
 
-    Acts as a convenience combination function of ``vapoursynth.core.text.Text`` and ``vapoursynth.core.std.Interleave``.
+    Acts as a convenience combination function of ``vapoursynth.core.text.Text`` and ``vapoursynth.core.std.Interleave`` .
 
-    :param label_alignment: Number from 1 to 9, corresponding to the positions of the keys on a numpad.
-                            Determines where to place clip name using ``vapoursynth.core.text.Text`` (Default: 7)
-    :param clips:           Keyword argument `name=clip` pairs for comparison. `name`s will be used to label the clips.
-                            The order the `clips` are specified will be respected in the comparison.
+    :param clips:
+    :param label_alignment:
     """
 
-    def __init__(self, label_alignment: Literal[1, 2, 3, 4, 5, 6, 7, 8, 9] = 7,
-                 **clips: vs.VideoNode):
-        super().__init__(label_alignment, **clips)
+    def __init__(self, clips: Union[Dict[str, vs.VideoNode], Sequence[vs.VideoNode]],
+                 /,
```

Comment:
I think I prefer `/` as trailing the positional args in the first line, if feasible, but that's just my opinion.

Reply:
In my opinion, it's clearer to have `/` on a line of its own when the args are already split over multiple lines. It gives a clear split of the "sections" at a glance, although that might not be necessary here as all sections individually fit in a single line.

Reply:
I would agree, if `clips` was not on the same line as `self` :wink: